straxen/plugins/acqmon_processing.py

```diff
@@ -55,161 +54,164 @@ def compute(self, raw_records_aqmon):
-        channel_numbers = np.arange(aq_ch_start, aq_ch_start + len(aqmon_channel_names))
-
-        # Keeping ch_range as class attribute for now, maybe will need it for other features
-        self.channel_range = self.config['channel_map']['aqmon']
-        self.channel_map = dict(zip(aqmon_channel_names, channel_numbers))
         self.veto_names = ['busy_', 'he_', 'hev_']
+        self.channel_map = {name: ch + straxen.n_hard_aqmon_start for ch, name in
+                            enumerate(['sum_wf', 'm_veto_sync',
+                                       'hev_stop', 'hev_start',
+                                       'he_stop', 'he_start',
+                                       'busy_stop', 'busy_start'])}
 
     def get_window_size(self):
         # Give a very wide window
-        return (self.config['max_veto_window'] * 100)
+        return int(self.config['max_veto_window'] * 100)
 
-    def compute(self, aqmon_hits):
+    def compute(self, aqmon_hits, start, end):
         hits = aqmon_hits
+        result = np.zeros(len(aqmon_hits) * len(self.veto_names), self.dtype)
+        vetos_seen = 0
 
-        res = defaultdict(list)
-        
-        for n, name in enumerate(self.veto_names):
+        for name in self.veto_names:
             veto_hits_start = channel_select_(hits, self.channel_map[name + 'start'])
             veto_hits_stop = channel_select_(hits, self.channel_map[name + 'stop'])
-            
-            # Here we rely on the fact that for each start, there is a single stop that 
-            # follows it in time. If this is not true, our hardware does not work. 
+
+            # Here we rely on the fact that for each start, there is a single stop that
+            # follows it in time. If this is not true, our hardware does not work.
             if len(veto_hits_start):
-                for t, time in enumerate(veto_hits_start['time']): 
+                for t, time in enumerate(veto_hits_start['time']):
                     # Find the time of stop_j that is closest to time of start_i
-                    inx = np.searchsorted(veto_hits_stop['time'], time, side = 'right') 
-                    
+                    inx = np.searchsorted(veto_hits_stop['time'], time, side='right')
+
                     if inx == len(veto_hits_stop['time']):
                         continue
                     else:
-                        res['veto_interval'].append(veto_hits_stop['time'][inx] - time)
-                        res["time"].append(time)
-                        res["endtime"].append(veto_hits_stop['time'][inx])
-                        res["veto_type"].append(name + 'veto')
-        
-        result = strax.dict_to_rec(res, self.dtype)
+                        result['veto_interval'][vetos_seen] = veto_hits_stop['time'][inx] - time
+                        result["time"][vetos_seen] = time
+                        assert time < veto_hits_stop['time'][inx]
```

Comment:
I agree (I thought I removed this assert statement)

Reply:
Besides the assert, my comment was also related to the actual things we would see.

I think I've already seen some AmBe runs where the number of busy stop and starts was not the same...maybe raising the AM threshold even more will help. Otherwise we'll have to think of some smarter way to deal with such cases in straxen.

Reply:
right, I agree, this was also one of the reasons why I thought using the `aqmon_hits` directly might be more robust as you are less depending on this